### ISSUE TITLE
IResolver docstrings contain broken references

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6338]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6338|
|Type|defect|
|Created|2013-02-28 14:08:52Z|
|Branch|https://github.com/twisted/twisted/tree/iresolver-epydocs-6338|

r37286 ([#4685](https://github.com/twisted/twisted/issues/4685)) introduced lots of new documentation but also many broken `C{Deferred}` references which should be `L{Deferred}` instead.

<details><summary>Searchable metadata</summary>

```
trac-id__6338 6338
type__defect defect
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_iresolver_epydocs_6338 branches/iresolver-epydocs-6338
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1362060532000000 1362060532000000
changetime__1362082820000000 1362082820000000
version__None None
owner__rwall rwall
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_1"></a>|@wallrj set status to `assigned`|
|-|-|

Ah. OK. Those @rtype C{Deferred} references were already there. I just moved them from t.n.client.lookup* to t.i.i.IResolver.lookup*

Should be easy to fix though. I'll do it now.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj commented|
|-|-|

(In [37308]) Branching to 'iresolver-epydocs-6338'

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_3"></a>|@wallrj removed owner<br>@wallrj set status to `new`|
|-|-|

Ready for review in log:branches/iresolver-epydocs-6338

 1. Replaces all C{Deferred} references with L{Deferred} references.
 2. Here are the apidocs built from this branch:
    https://buildbot.twistedmatrix.com/builds/apidocs/apidocs-37309.tar.bz2
 3. And I started a
    [http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/iresolver-epydocs-6338 build]

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @wallrj|
|-|-|

Looks good. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set status to `closed`|
|-|-|

(In [37323]) Merge iresolver-epydocs-6338: Fix epydoc links to Deferred in IResolver

Author: rwall
Reviewer: tom.prince
Fixes: [#6338](https://github.com/twisted/twisted/issues/6338)

 * Replace C{Deferred} with L{Deferred} references in IResolver docstrings.
